includes/class-sensei-course-structure.php

```diff
@@ -97,9 +102,26 @@ public function get( $context = 'view' ) {
 			$structure[] = $this->prepare_lesson( $lesson );
 		}
 
+		if ( $no_cache ) {
+			remove_filter( 'posts_where', [ $this, 'filter_no_cache_where' ] );
+		}
+
 		return $structure;
 	}
 
+	/**
+	 * Filter where adding an extra condition to avoid cache.
+	 *
+	 * @access private
+	 *
+	 * @param string $where Current where.
+	 *
+	 * @return string Where with extra condition to avoid cache.
+	 */
+	public function filter_no_cache_where( $where ) {
+		return $where . ' AND ' . time() . ' = ' . time();
```

Comment:
I don't think this would ever happen, but I wonder if it is a tiny-tiny-tiny bit safer to set `time()` to a variable here. I mean, we'd have to be really unlucky and I doubt PHP would ever do that.

Reply:
Ah, of course! Fixed here: https://github.com/Automattic/sensei/pull/4236/commits/c0fffe10d244900dee3ac9f7d3926b8da64ee2cd